### PR TITLE
Launch Management Chain for E2E

### PR DESCRIPTION
**Pull request by ronnno**
launch a second virtual chain for e2e tests:
1. in-process e2e harness - vcid 40 for mgmt chain
2. docker based e2e -  - vcid 40 for new chain
3. chain-on-pr - launch vcid: `100000 + (prNumber * 2)` for mgmt chain and `100000 + (prNumber * 2 + 1)` for app chain
4. testnet e2e - launch vcid 40 for consensus benchmark tests and 2011 for lean helix 

for in-process e2e harness:
- public API ports used (where n is the network size):
  - app chain ports for each node:`8090, 8090+1,...,8090+(n-1)` 
  - mgmt chain ports for each node:`8090+n, 8090+(n+1),...,8090(2n-1)` 
- both processor artifact path (`test/e2e/_tmp/`contract compiler work dir) and virtual chain data dir will be appended by a vchain id component of the form: `vcid_[vcid]`

### CODE REVIEW

**Comment by ronnno** [comment]
race test falls on LH race condition, fixed in https://github.com/orbs-network/lean-helix-go/pull/70

**Comment by ronnno** [comment]
@itamararjuan need your help with ci/e2e - `docker/test/test.sh` says:

```
rm: cannot remove ‘_tmp/blocks/node4-mgmt/blocks’: Permission denied
rm: cannot remove ‘_tmp/blocks/node3-mgmt/blocks’: Permission denied
rm: cannot remove ‘_tmp/blocks/node2-mgmt/blocks’: Permission denied
rm: cannot remove ‘_tmp/blocks/node1-mgmt/blocks’: Permission denied
```

File: .circleci/config.yml [hunk 1]
```diff
@@ -127,7 +127,7 @@ jobs:
 
   deploy_testnet:
     docker:
-      - image: circleci/node:11.2.0
+      - image: itamararjuan/golang_with_node_aws
```
Comment (electricmonk) [inline .circleci/config.yml]: @itamararjuan why are you using your own image?

File: config/system_factory_presets.go [hunk 2]
```diff
@@ -115,21 +115,11 @@ func ForProduction(processorArtifactPath string) mutableNodeConfig {
...
-	nodePrivateKey primitives.EcdsaSecp256K1PrivateKey,
-	gossipPeers map[string]GossipPeer,
-	genesisValidatorNodes map[string]ValidatorNode,
-	blockStorageDataDirPrefix string,
-	processorArtifactPath string,
-	ethereumEndpoint string,
-	constantConsensusLeader primitives.NodeAddress,
-	activeConsensusAlgo consensus.ConsensusAlgoType,
-) NodeConfig {
+func ForE2E(httpAddress string, virtualChainId primitives.VirtualChainId, gossipListenPort int, nodeAddress primitives.NodeAddress, nodePrivateKey primitives.EcdsaSecp256K1PrivateKey, gossipPeers map[string]GossipPeer, genesisValidatorNodes map[string]ValidatorNode, blockStorageDataDirPrefix string, processorArtifactPath string, ethereumEndpoint string, constantConsensusLeader primitives.NodeAddress, activeConsensusAlgo consensus.ConsensusAlgoType) NodeConfig {
```
Comment (electricmonk) [inline config/system_factory_presets.go]: please unfmt

File: docker/test/docker-compose.yml [hunk 3]
```diff
@@ -51,7 +51,7 @@ services:
         ipv4_address: 192.168.199.12
 
 
-  orbs-network-node-1:
+  orbs-app-chain-node-1:
```
Comment (electricmonk) [inline docker/test/docker-compose.yml]: this file has become long and repetitive; please see if we can simplify it (separate PR)

File: test/e2e/harness.go [hunk 4]
```diff
@@ -175,18 +186,7 @@ func (h *harness) waitUntilTransactionPoolIsReady(t *testing.T) {
...
-		primitives.NodeAddress{},
-		primitives.EcdsaSecp256K1PrivateKey{},
-		nil,
-		nil,
-		"",
-		"",
-		"",
-		primitives.NodeAddress{},
-		0).
+	return config.ForE2E("", 0, 0, primitives.NodeAddress{}, primitives.EcdsaSecp256K1PrivateKey{}, nil, nil, "", "", "", primitives.NodeAddress{}, 0).
```
Comment (electricmonk) [inline test/e2e/harness.go]: please unfmt

File: docker/test/docker-compose.yml (hunk 3 above)
Comment (ronnno) [reply docker/test/docker-compose.yml]: opened #1318

**Pull request merged**